ReactApp/car_rental_react/src/routes/CarReserve.js

```diff
@@ -1,154 +1,121 @@
 import { Button } from "reactstrap";
 import axios from "axios";
+import { useLocation } from "react-router-dom";
+import { useAuth } from "../provider/authContext";
 
 const CarReserve = () => {
-  function signIn() {
-    alert("Feature not yet implemented!");
-  }
+  //const { isSignedIn, user, logOut } = useAuth();
+  //console.log(isSignedIn, user);
 
-  function submitRental() {
-    // first create a customer from the details inputted
-    const customer_fields = document.getElementsByClassName("rental-input");
-    var customer_input = [];
-    for (let index = 0; index < customer_fields.length; index++) {
-      customer_input.push(customer_fields[index].value);
-    }
+  // get info from car selection
+  const location = useLocation();
+  console.log(location.state);
+
+  //get car information
+  var car_model;
+  var car_type;
+  var car_id;
+  axios
+    .get("/api/cars/" + location.state.carid + "/")
+    .then((res) => {
+      // get model of car
+      car_model = res.data.manufacturer + " " + res.data.model;
+      let element = document.getElementById("cartype");
+      element.innerHTML = "Car Type: " + car_model;
+
+      // get car type and car id
+      car_type = res.data.car_type;
+      car_id = res.data.car_id;
+
+      // update estimated price
+      let price_element = document.getElementById("price");
+      price_element.innerHTML =
+        "Estimated Price: " + location.state.estimatedcost + "$";
+    })
+    .catch((err) => console.log(err));
 
-    const customer_details = {
-      first_name: customer_input[0],
-      last_name: customer_input[1],
-      drivers_license: customer_input[10],
-      email: customer_input[2],
-      DOB: customer_input[4],
-      gold_member: false,
-      province: customer_input[7],
-      city: customer_input[9],
-      postal_code: customer_input[8],
-      street_number: customer_input[5],
-      street_name: "ST",
-      unit_number: customer_input[6],
+  //get return branch information
+  var branch_to;
+  axios
+    .get("/api/branches/" + location.state.branchgoesto + "/")
+    .then((res) => {
+      // set return branch name and update summary
+      branch_to = res.data.street_number + " " + res.data.street_name;
+      let element = document.getElementById("rbranch");
+      element.innerHTML = "Return Branch: " + branch_to;

```

Comment:
Same here, can you use state instead of changing innerHTML? This will improve performance as well as telling React to keep track of states.